presto-singlestore/src/main/java/com/facebook/presto/plugin/singlestore/SingleStoreClient.java

```diff
@@ -144,21 +149,36 @@ protected String toSqlType(Type type)
             if (varcharType.isUnbounded()) {
                 return "longtext";
             }
-            if (varcharType.getLengthSafe() <= 255) {
-                return "tinytext";
+            if (varcharType.getLengthSafe() <= 21844) {
+                // 21844 is the maximum length a singlestore varchar supports.
+                return super.toSqlType(type);
             }
-            if (varcharType.getLengthSafe() <= 65535) {
-                return "text";
-            }
-            if (varcharType.getLengthSafe() <= 16777215) {
+            if (varcharType.getLengthSafe() <= 5592405) { // 16MB
                 return "mediumtext";
             }
-            return "longtext";
+            if (varcharType.getLengthSafe() <= 1431655765) { // 100MB to 1GB
+                return "longtext"; // max = 1431655765
+            }
+            else {
+                throw new PrestoException(NOT_SUPPORTED, "Unsupported column width: " + varcharType.getLengthSafe());
+            }
         }
 
         return super.toSqlType(type);
     }
 
+    @Override
+    public Optional<ReadMapping> toPrestoType(ConnectorSession session, JdbcTypeHandle typeHandle)
```

Comment:
Small change needed here with https://github.com/prestodb/presto/pull/25437/ merged in